pkgs/build-support/go/module.nix

```diff
@@ -167,7 +167,7 @@ let
         export GOPROXY=file://${goModules}
       '' else ''
         rm -rf vendor
-        cp -r --reflink=auto ${goModules} vendor
+        ln -s ${goModules} vendor
```

Comment:
> Go doesn't need to modify this

go may not need to modify it but that doesn't mean that the package itself won't modify it.

We used to use `ln` and we switched to `cp`: https://github.com/NixOS/nixpkgs/commit/66d424514f2dff8968e41dfd70c8836f995fb7be

Reply:
Maybe it could be a `buildGoModule` option? `vendorRW = true;` or something?

Reply:
`buildGoModule` does not need more options.

Reply:
Could use rsync to speed up the copy?

Reply:
Reflink might be actually faster then rsync.

Reply:
Ah, I might be misremembering, `gomod2nix` uses rsync but maybe it is for a different reason.

Reply:
Motivation was this: https://github.com/numtide/build-go-cache
But than I guess there is not much point in upstreaming this since buildGoModule will stay slow even after adding the cache.